cmd/kops/update_cluster.go

```diff
@@ -290,19 +299,29 @@ func RunUpdateCluster(ctx context.Context, f *util.Factory, out io.Writer, c *Up
 		return nil, err
 	}
 
+	minControlPlaneRunningVersion := cluster.Spec.KubernetesVersion
+	if !c.IgnoreVersionSkew {
+		minControlPlaneRunningVersion, err = checkControlPlaneRunningVersion(ctx, cluster.ObjectMeta.Name, minControlPlaneRunningVersion)
+		if err != nil {
+			klog.Warningf("error checking control plane running verion: %v", err)
```

Comment:
I think we should just return the error, unless there's some reason to expect this error to happen which we need to tolerate

Reply:
I agree we shouldn't log this warning when we know the check wont succeed otherwise it will cause confusion. For example, the e2e tests show this log line:

`W1202 14:53:23.985493   47436 update_cluster.go:335] error checking control plane running verion: cannot load kubecfg settings for "e2e-pr16932.pull-kops-e2e-k8s-aws-calico.test-cncf-aws.k8s.io": context "e2e-pr16932.pull-kops-e2e-k8s-aws-calico.test-cncf-aws.k8s.io" does not exist`

Another failure case would be if the kubeconfig context _does_ exist but the k8s API isn't up - perhaps a user is recreating a previously-deleted cluster. In that case it may be more acceptable to log the warning.

Perhaps we could also avoid confusion by clarifying the impact:

```suggestion
			klog.Warningf("error checking control plane running version, assuming no k8s upgrade in progress: %v", err)
```